cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -457,6 +465,13 @@ public static KafkaCluster fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup vers
         }
 
         KafkaConfiguration configuration = new KafkaConfiguration(kafkaClusterSpec.getConfig().entrySet());
+        // If  required Cruise Control metric reporter configurations are missing set them using Kafka defaults
+        if (configuration.getConfigOption(CC_NUM_PARTITIONS_CONFIG_FIELD) == null) {
+            result.ccDefaultNumPartitions = configuration.getConfigOption(KAFKA_NUM_PARTITIONS_CONFIG_FIELD, result.kafkaDefaultNumPartitions);
+        }
+        if (configuration.getConfigOption(CC_REPLICATION_FACTOR_CONFIG_FIELD) == null) {
+            result.ccDefaultReplicationFactor = configuration.getConfigOption(KAFKA_REPLICATION_FACTOR_CONFIG_FIELD, result.kafkaDefaultReplicationFactor);
+        }
```

Comment:
If I read this section right, the flow is that:
1) The user specifies these topions in `Kafka.spec.kafka.config`
2) You read them from there
3) You pass them to the KafkaBrokerConfigurationBuilder which puts them into the broker configuration file

But at the same time, they will be put into the KAfka broker configuration as part of `withUserConfiguration` so they will be there twice, or?

Reply:
This is the intended logic flow:
1) The user specifies options, `cruise.control.metrics.topic.num.partitions` and/or `cruise.control.metrics.topic.replication.factor` in the `Kafka.spec.kafka.config`
2) These options are read
2) The if-statements highlighted above will not execute since they are found in the map with getConfigOption(), so the fields `ccNumPartitions` and `ccReplicationFactor` will remain null
3)  The `ccNumPartitions` and `ccReplicationFactor` fields are passed to  KafkaBrokerConfigurationBuilder, but will not be added to the broker configuration since they are null. This way these options are only added once to the broker configuration.
https://github.com/strimzi/strimzi-kafka-operator/blob/2a441eb0d9ece5350e54c33f9a34dcb0f284fb42/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java#L90-L95